app/src/main/java/net/skyscanner/backpack/demo/ComponentDetailFragment.kt

```diff
@@ -17,25 +15,21 @@ import net.skyscanner.backpack.demo.data.ComponentRegistry
  */
 open class ComponentDetailFragment : Fragment() {
 
-    override fun onCreate(savedInstanceState: Bundle?) {
-        super.onCreate(savedInstanceState)
-
-        if (arguments!!.containsKey(ARG_ITEM_ID)) {
-
-            val activity = this.activity
-            val appBarLayout = activity!!.findViewById<CollapsingToolbarLayout>(R.id.toolbar_layout)
-            if (appBarLayout != null) {
-                appBarLayout.title = arguments!!.getString(ARG_ITEM_ID)
-            }
-        }
+  override fun onCreate(savedInstanceState: Bundle?) {
+    if (arguments!!.containsKey(ARG_ITEM_ID)) {
+      val activity = this.activity as ComponentDetailActivity
+      activity.title = arguments!!.get(ARG_ITEM_ID) as CharSequence?
```

Comment:
why is the cast to CharSequence necessary? doesn't it have a `getString/Text` function?